### PR TITLE
DLL pull request

### PR DESCRIPTION
**Pull request by WildandArt**
Impl of doubly linked list

### CODE REVIEW

File: README.md
Comment (TalAharonSQream) [inline README.md:1]: Overview of the project before table of content. 
Whu java 22? is there a feature that you used that is only in java 22? java 17 users won't be able to compile the code with the javac 17? please verify that and fix the readme to be more generic about the Jave version the user needs. if the user already have jdk installed instruct him, how to move to a higher version if he needs. 


Explanation of the functions and methods is good but is looks a bit monotonic on the eyes so please make it colorful and use bold and different text sizes to emphasize every description and make space between one another.

YOu don't need to write about TDD, this is a method you use to develop you code but not relevant to the project description aka readme. no one cares how you wrote the test and in what way what they need to know that there are tests and that they are working and they can count on them. please add a description on how to run the tests.

File: pom.xml
Comment (TalAharonSQream) [inline pom.xml:1]: make sure you have everything you need and no unnecessary things

File: src/main/java/com/DoublyLinkedList/App.java [hunk 1]
```diff
@@ -1,13 +1,9 @@
...
-/**
- * Hello world!
- *
- */
 public class App 
 {
     public static void main( String[] args )
     {
-        System.out.println( "Hello World!" );
+        //with tests we won't need main function
```
Comment (TalAharonSQream) [inline src/main/java/com/DoublyLinkedList/App.java]: Add a space "// text"

also i would just leave a comment empty and that's it no need to specify why

File: src/main/java/com/DoublyLinkedList/DoublyLinkedList.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
+
+/**
+ * @package com.DoublyLinkedList
+ * @brief This package contains the implementation of a doubly linked list.
+ */
+
+ package com.DoublyLinkedList;
+
+//  import com.DoublyLinkedList.Impl.NodeImpl;
```
Comment (TalAharonSQream) [inline src/main/java/com/DoublyLinkedList/DoublyLinkedList.java]: remove if not used.

File: src/main/java/com/DoublyLinkedList/Impl/DoublyLinkedListImpl.java
Comment (TalAharonSQream) [inline src/main/java/com/DoublyLinkedList/Impl/DoublyLinkedListImpl.java:1]: The implementation looks good but there is a lot of if statements that in the eyes bother me. try to reduce the number of if statements while keeping the code work as expected and pass tests. there is a lot of methods on how to reduce if statements read about them.

File: src/main/java/com/DoublyLinkedList/Node.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
+
+// package com.DoublyLinkedList;
```
Comment (TalAharonSQream) [inline src/main/java/com/DoublyLinkedList/Node.java]: remove unused comments from code

**Review by TalAharonSQream — CHANGES_REQUESTED** [review]

File: README.md
Comment (WildandArt) [reply README.md:1]: compressed tdd talk to two lines. fixed to 17, it works with 17 too. put all method names inside code brackets, now its more pleasant to the eye

File: src/main/java/com/DoublyLinkedList/App.java (hunk 1 above)
Comment (WildandArt) [reply src/main/java/com/DoublyLinkedList/App.java]: fixed

File: src/main/java/com/DoublyLinkedList/DoublyLinkedList.java (hunk 2 above)
Comment (WildandArt) [reply src/main/java/com/DoublyLinkedList/DoublyLinkedList.java]: ok

File: src/main/java/com/DoublyLinkedList/Impl/DoublyLinkedListImpl.java
Comment (WildandArt) [reply src/main/java/com/DoublyLinkedList/Impl/DoublyLinkedListImpl.java:1]: fixed, review again

File: src/main/java/com/DoublyLinkedList/Node.java (hunk 3 above)
Comment (WildandArt) [reply src/main/java/com/DoublyLinkedList/Node.java]: fixed

File: README.md
Comment (YamtalDev) [reply README.md:1]: nice!

**Review by YamtalDev — APPROVED** [review]

**Review by TalAharonSQream — APPROVED** [review]

**Comment by WildandArt** [comment]
Fixed version of DLL

**Pull request merged**